### DISCUSSION TITLE
nix flake part error: The option `removeReferencesTo' does not exist.

### DISCUSSION THREAD

**Discussion by TristanCacqueray** [srid/emanote#527] [Q&A]
**Describe the bug**
Hello, after updating to the latest emanote (through https://github.com/TristanCacqueray/TristanCacqueray.github.io/commit/262159de64ca95558c6ca69e43f8baa12a052d6f) I get the following exception:

```ShellSession
$ nix run github:TristanCacqueray/TristanCacqueray.github.io/ema-part-error
error:
       … while evaluating the attribute 'value'

         at /nix/store/lrmxkfadhlcrakza60w0mc8sc327a5qg-source/lib/modules.nix:809:9:

          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       … while calling the 'addErrorContext' builtin

         at /nix/store/lrmxkfadhlcrakza60w0mc8sc327a5qg-source/lib/modules.nix:809:17:

          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |                 ^
          810|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: The option `removeReferencesTo' does not exist. Definition values:
       - In `/nix/store/mfalmay99kyazcbyp7rhm7pxq1k6xs2p-source/flake.nix, via option perSystem, via option perSystem.x86_64-linux.haskellProjects.default.settings.emanote':
           [
             <derivation pandoc-3.0.1>
             <derivation pandoc-3.1.11>
             <derivation pandoc-types-1.23.1>
             <derivation warp-3.3.25>
           ...
```

It seems to be related to the flake part. It's not the first time the flake part caused problems, in the past I was able to resolve them by forcing a flake update, but this time I can't figure it out. For what it's worth, I fixed this issue by using standalone commands in https://github.com/TristanCacqueray/TristanCacqueray.github.io/commit/3dba63a54bf5af42f2f732a7fb3c34b58c20bf2b .

**Comment by srid** [comment]
> Hello, after updating to the latest emanote (through [TristanCacqueray/TristanCacqueray.github.io@262159d](https://github.com/TristanCacqueray/TristanCacqueray.github.io/commit/262159de64ca95558c6ca69e43f8baa12a052d6f)) I get the following exception:

But this comment has no corresponding update to the `flake.lock` file?

**Reply by srid** [reply]
You want to run `nix flake lock` before commiting.

**Reply by TristanCacqueray** [reply]
I did tried that too, forgot to add to the repro branch. Here I commited the output of flake lock, still the same error:

```ShellSession
$  nix run github:TristanCacqueray/TristanCacqueray.github.io/b281f09c11ca9ecdbdcc8f93d500aae0ca859652
error:
       … while evaluating the attribute 'value'
```

**Reply by srid** [reply]
Okay, try `nix flake update emanote`

**Reply by srid** [reply]
By the way, for Nix questions I recommend https://nixos.zulipchat.com/

It is maintained by us!

**Reply by TristanCacqueray** [reply]
Still the same error after `nix flake update emanote`: https://github.com/TristanCacqueray/TristanCacqueray.github.io/commit/e235d7d57db2275b958ea745e1146fe779e36867 .

I meant, I actually prefer my solution without flake-part since it's easier to work with, I am mostly reporting this for your information.

**Reply by srid** [reply]
> Still the same error

Are you sure it is the *same* error?

**Comment by srid** [comment]
Fix: https://github.com/srid/emanote-template/commit/d02c989a7c72d8bee7c5dde2e9f6c8f0110ab8bf#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R23
